marimo/_ast/visitor.py

```diff
@@ -281,20 +287,42 @@ def generic_visit(self, node: ast.AST) -> None:
             # Other nodes that don't introduce a new scope
             super().generic_visit(node)
 
+    def _visit_and_get_refs(self, node: ast.AST) -> set[Name]:
+        self.ref_stack.append(set())
+        self.generic_visit(node)
+        refs = self.ref_stack.pop()
+        self.ref_stack[-1].update(refs)
+        return refs
```

Comment:
I kind of understand what's happening here. But I don't understand the point of 

```python
self.ref_stack[-1].update(refs)
```

I know I'm just misunderstanding something -- can you add a comment explaining what's happening?